pkgs/development/ocaml-modules/dose3/default.nix

```
@@ -0,0 +1,62 @@
+{ lib, buildDunePackage, fetchFromGitLab
+, camlzip, ocamlgraph, parmap, re, stdlib-shims
+, base64, bz2, ocaml_extlib, cudf
+, dpkg, odoc, ounit, python39, python39Packages
+}:
+
+buildDunePackage rec {
+  pname = "dose3";
+  version = "6.1";
+
+  src = fetchFromGitLab {
+    owner = "irill";
+    repo = "dose3";
+    rev = version;
+    sha256 = "sha256-RFChY7VH2pVD4u5U1qng46h9aAv8I/3yXVaNmFDmKFI=";
+  };
+
+  minimalOCamlVersion = "4.03";
+  useDune2 = true;
+
+  buildInputs = [
+    ounit
+    base64
+    cudf
+    parmap
+    re
+  ];
+  propagatedBuildInputs = [
+    bz2
+    camlzip
+    ocaml_extlib
+    ocamlgraph
+    stdlib-shims
+  ];
+
+  nativeCheckInputs = [
+    python39                  # Replaces: conf-python-3
+    python39Packages.pyyaml   # Replaces: conf-python3-yaml
+    dpkg                      # Replaces: conf-dpkg
+  ];
+  checkInputs = [
+    odoc
+    ounit
+  ];
+  doChecks = true;
```

```suggestion
  doCheck = true;
```

Also you may want to enable tests only when `ounit` is available (i.e., `lib.versionAtLeast ocaml.version "4.04"`).

---

Thanks. Now it breaks on [tests](https://gitlab.com/irill/dose3/-/blob/master/src/applications/dune):
`dose-tests alias src/applications/runtest (exit 127)`, will investigate it.

---

> There is some pending TODO in the `cudf` derivation.

Just enabled test for `cudf`. Thanks.

---

I have extended the passthru.tests to other dependencies. I believe these tests are not run when building, only on CI. So it is less of problem to do this. As always let me know if I did something stupid. :-)

And yes, tests are still failing. =/

---

Current errors:
```
runTest (__main__.DoseTests)
Test : exitcode_distcheck_64_notexistent
Group : distcheck
Cmd : ./src/applications/distcheck.exe --checkonly a -f -e cudf://notexistent
Expected file : /dev/null
Expected exitcode : 64
 ... Fatal error in module dose_common.input: 
 Input file notexistent does not exist.
```
---
```
runTest (__main__.DoseTests)
Test : exitcode_distcheck_64_parse_error
Group : distcheck
Cmd : ./src/applications/distcheck.exe --checkonly a -f -e cudf://tests/DebianPackages/Packages_20201226T144309Z_amd64.xz
Expected file : /dev/null
Expected exitcode : 64
 ... Fatal error in module doseparse.stdLoaders: 
 Error while parsing CUDF from tests/DebianPackages/Packages_20201226T144309Z_amd64.xz (line: 1): unexpected RFC 822 token
```

---

Is it possible to skip these?

---

I have disabled tests. Made an attempt of partial tests but didn't work.